dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2127,166 @@ def _update_chi2(self):
+                )
+                added_true_count = added_sample_size - added_null_count
+
+                prev_class_means = self._nan_replication_metrics[col_id]["class_mean"]
+                prev_mean_not_nan, prev_mean_nan = prev_class_means
+
+                mean_not_nan = utils.combine_means(
+                    prev_mean_not_nan, mean_not_nan, prev_true_count, added_true_count
+                )
+                mean_nan = utils.combine_means(
+                    prev_mean_nan, mean_nan, prev_null_count, added_null_count
+                )
+
+            col_name = getattr(self._profile[col_id], "name")
+            self._nan_replication_metrics[col_id] = {
+                "column_name": col_name,
+                # Array index serves as class label
+                # 0 indicates not NaN, 1 indicates NaN
+                "class_prior": [prior_not_nan, prior_nan],
+                "class_mean": [mean_not_nan, mean_nan],
+            }
+
+    def _merge_nan_replication_metrics(self, other):
+        """
+        Merge NaN replication metrics between two data profiles
+
+        :param other: profile being added to this one.
+        :type other: StructuredProfiler
+        :return: merged nan replication metrics
+        :rtype: dict
+        """
+
+        if (
+            not self.options.synthetic_data.replicate_nan.is_enabled
+            or not other.options.synthetic_data.replicate_nan.is_enabled
+        ):
+            return {}
+
+        merged_properties = {}
+        for col_id in range(len(self._profile)):
+            self_null_count = getattr(self._profile[col_id], "null_count")
+            other_null_count = getattr(other._profile[col_id], "null_count")
+            null_count = self_null_count + other_null_count
+            if null_count == 0:
+                continue
+
+            merged_properties[col_id] = {}
+            self_sample_size = getattr(self._profile[col_id], "sample_size")
+            other_sample_size = getattr(other._profile[col_id], "sample_size")
+            sample_size = self_sample_size + other_sample_size
+            true_count = sample_size - null_count
+
+            prior_nan = null_count / sample_size
+            prior_not_nan = true_count / sample_size
+
+            merged_properties[col_id]["class_prior"] = [prior_not_nan, prior_nan]
+
+            col_name = getattr(self._profile[col_id], "name")
+            merged_properties[col_id]["column_name"] = col_name
+
+            # 'if null_count == 0' check guarantees that either self or other has _nan_replication_metrics for col_id
+            if col_id not in self._nan_replication_metrics:
+                other_mean = other._nan_replication_metrics[col_id]["class_mean"]
+                merged_properties[col_id]["class_mean"] = other_mean
+
+            elif col_id not in other._nan_replication_metrics:
+                self_mean = self._nan_replication_metrics[col_id]["class_mean"]
```

Comment:
`self._nan_replication_metrics[col_id]["class_mean"]` repated...

`self._nan_replication_metrics[col_id]["class_mean"]` --> `class_mean = self._nan_replication_metrics[col_id]["class_mean"]` and then just use `class_mean` in the rest of the function